store/src/java/com/zimbra/cs/service/account/Auth.java

```diff
@@ -508,4 +528,36 @@ public static void addAccountToLogContextByAuthToken(Provisioning prov, AuthToke
         if (aid != null && !aid.equals(id))
             AccountUtil.addAccountToLogContext(prov, aid, ZimbraLog.C_ANAME, ZimbraLog.C_AID, null);
     }
+
+    public static AuthToken validateAndCreateNewJwtToken(String jwt, Account acct, Provisioning prov) throws ServiceException {
+        if (StringUtil.isNullOrEmpty(jwt)) {
+            throw ServiceException.AUTH_REQUIRED();
+        }
+        String tokenKey = acct.getJWTKey();
```

Comment:
Rest of the changes are done but this change is done under https://github.com/Zimbra/zm-mailbox/pull/317 by @shrikant-prasad 
Will update pr once this pr is closed, as the changes are dependent.